### DISCUSSION TITLE
http://localhost:5000 Requests Username and Password?

### DISCUSSION THREAD

**Discussion by TheWaterbug** [mrlt8/docker-wyze-bridge#1253] [Q&A]
I had run Docker-Wyze-Bridge on this Win10 box a few months ago successfully (pre-Wyze-2FA requirement), but then turned it off because I wanted to run it on a different box. I've changed my mind again, and now I want to run on this box again, so I went to Wyze to turn on my 2FA and get my API key, then did:

docker run -p 8554:8554 -p 8888:8888 -p 5000:5000 mrlt8/wyze-bridge

and Docker Desktop dutifully launched and reported that the container was running. I went to:

http://localhost:5000

which then asked me for my Wyze credentials, which I entered:

![image](https://github.com/mrlt8/docker-wyze-bridge/assets/78178387/7a0c7999-2113-4e26-a7fa-a71f4f5e5176)

After I clicked Login I got:

![image](https://github.com/mrlt8/docker-wyze-bridge/assets/78178387/34609f7e-a7f3-4bdc-8eb8-e7b341dc483f)

and then this:

![image](https://github.com/mrlt8/docker-wyze-bridge/assets/78178387/f45a7761-0984-4585-90d5-920562ce9b3a)

What username and password is it asking for? I've tried my Wyze user/pass, my Windows user/pass, and neither of them works.

Powershell reports:

`docker run -p 8554:8554 -p 8888:8888 -p 5000:5000 mrlt8/wyze-bridge

DEFAULT WB_PASSWORD:
password='xxxxxxxxxx'

DEFAULT WB_API:
password='xxxxxxxxxxxx'

🚀 DOCKER-WYZE-BRIDGE v2.9.6 X86_64

[WyzeBridge] [WARN] Credentials are NOT set
[WyzeBridge] 🔍 Could not find local cache for 'auth'
[WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
[WyzeBridge] Credentials required to complete login!
 * Serving Flask app 'frontend'
[WyzeBridge] Please visit the WebUI to enter your credentials.
 * Debug mode: off
[WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://172.17.0.2:5000
[WyzeBridge] Press CTRL+C to quit
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:10:27] "GET / HTTP/1.1" 302 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:10:27] "GET /login HTTP/1.1" 200 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:10:27] "GET /static/bulma.css HTTP/1.1" 304 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:10:27] "GET /static/site.css HTTP/1.1" 304 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:10:27] "GET /static/bulma-toast.js HTTP/1.1" 304 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:10:27] "GET /favicon.ico HTTP/1.1" 404 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:10:44] "POST /login HTTP/1.1" 200 -
[WyzeBridge] 💾 Saving 'auth' to local cache...
[WyzeBridge] 🔍 Could not find local cache for 'user'
[WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
[WyzeBridge] 💾 Saving 'user' to local cache...
[WyzeBridge] 🔍 Could not find local cache for 'cameras'
[WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
[WyzeBridge] [API] Fetched [12] cameras
[WyzeBridge] 💾 Saving 'cameras' to local cache...
[WyzeBridge] [+] Adding WyzeCamv3Pro [HL_CAM3P]
[WyzeBridge] [+] Adding BananaCam2 [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding SmallGarage [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding Corn Seeds [WYZEC1-JZ]
[WyzeBridge] [+] Adding Eastvale Garage [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding Exultant Garage [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding Mom & Dad’s TV Cam [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding Breadbox  [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding BananaCam [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding HummingbirdCam [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding Finch Cam [WYZE_CAKP2JFUS]
[WyzeBridge] [+] Adding RandomCam [WYZEC1-JZ]
[WyzeBridge] starting MediaMTX 1.1.1
[WyzeBridge] 🎬 12 streams enabled
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:10:49] "GET / HTTP/1.1" 401 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:14:20] "GET / HTTP/1.1" 401 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:14:31] "GET / HTTP/1.1" 401 -
[WyzeBridge] 172.17.0.1 - - [07/Jun/2024 18:14:39] "GET / HTTP/1.1" 401 -`

where the last 3 messages appeared after I attempted and failed to log in. The messages about Adding the Cameras suggests that I have entered my Wyze credentials successfully. 

If I Delete Forever the container and start over, I get the exact same symptoms. Is there something else I need to configure and/or different credentials to enter?

Thanks!

**Comment by mrlt8** [comment]
The WebUI will use a random password if you start the container without the wyze credentials. You should be able to login by using `wbadmin` as the username and the `WB_PASSWORD` that was displayed at startup:
```
DEFAULT WB_PASSWORD:
password='xxxxxxxxxx'
```

You can change the WebUI credentials by setting `WB_USERNAME` and `WB_PASSWORD` or disable authentication by setting `WB_AUTH=False`.

Streams and API endpoints will also require authentication when `WB_AUTH` is enabled. 
These will use `wb` as the username and a random `WB_API` key that will be generated and displayed in the logs and also visible in the WebUI. You can also customize this by setting `WB_API`.

**Comment by foudafoudaongolo-creator** [comment]
from flask import Flask, render_template, request, redirect, url_for, send_file
import sqlite3
import pandas as pd
from datetime import datetime
from reportlab.pdfgen import canvas # pip install reportlab pandas

app = Flask(__name__)

DB = 'salaires.db'

# 1. INITIALISER LA BASE
def init_db():
    conn = sqlite3.connect(DB)
    c = conn.cursor()
    c.execute('''CREATE TABLE IF NOT EXISTS fonctionnaires
                 (id INTEGER PRIMARY KEY, matricule TEXT, nom TEXT, prenom TEXT,
                 departement TEXT, grade TEXT, salaire_brut REAL)''')
    conn.commit()
    conn.close()

# 2. FONCTION CALCUL SALAIRE
def calculer_salaire_net(salaire_brut):
    cnps = salaire_brut * 0.042 # 4.2%
    irpp = salaire_brut * 0.10 # 10% simplifié
    salaire_net = salaire_brut - cnps - irpp
    return round(salaire_net, 2), round(cnps, 2), round(irpp, 2)

@app.route('/')
def index():
    conn = sqlite3.connect(DB)
    df = pd.read_sql_query("SELECT * FROM fonctionnaires", conn)
    conn.close()

    # Calcul masse salariale par département
    masse_par_dept = df.groupby('departement')['salaire_brut'].sum().to_dict() if not df.empty else {}
    total_masse = df['salaire_brut'].sum() if not df.empty else 0

    return render_template('index.html', fonctionnaires=df.to_dict('records'),
                           masse=masse_par_dept, total=total_masse)

# 3. AJOUTER FONCTIONNAIRE
@app.route('/ajouter', methods=['POST'])
def ajouter():
    data = request.form
    conn = sqlite3.connect(DB)
    c = conn.cursor()
    c.execute("INSERT INTO fonctionnaires (matricule, nom, prenom, departement, grade, salaire_brut) VALUES (?,?,?,?,?,?)",
              (data['matricule'], data['nom'], data['prenom'], data['departement'], data['grade'], float(data['salaire_brut'])))
    conn.commit()
    conn.close()
    return redirect(url_for('index'))

# 4. GÉNÉRER BULLETIN PDF
@app.route('/bulletin/<int:id>')
def bulletin(id):
    conn = sqlite3.connect(DB)
    c = conn.cursor()
    c.execute("SELECT * FROM fonctionnaires WHERE id=?", (id,))
    f = c.fetchone()
    conn.close()

    salaire_net, cnps, irpp = calculer_salaire_net(f[6])

    # Créer PDF
    filename = f"Bulletin_{f[2]}_{f[3]}.pdf"
    c = canvas.Canvas(filename)
    c.setFont("Helvetica-Bold", 16)
    c.drawString(100, 800, "BULLETIN DE PAIE")
    c.setFont("Helvetica", 12)
    c.drawString(50, 760, f"Nom: {f[2]} {f[3]}")
    c.drawString(50, 740, f"Matricule: {f[1]}")
    c.drawString(50, 720, f"Département: {f[4]}")
    c.drawString(50, 700, f"Grade: {f[5]}")
    c.drawString(50, 670, f"Salaire Brut: {f[6]} FCFA")
    c.drawString(50, 650, f"CNPS 4.2%: -{cnps} FCFA")
    c.drawString(50, 630, f"IRPP 10%: -{irpp} FCFA")
    c.setFont("Helvetica-Bold", 14)
    c.drawString(50, 600, f"SALAIRE NET: {salaire_net} FCFA")
    c.save()

    return send_file(filename, as_attachment=True)

# 5. EXPORT EXCEL
@app.route('/export')
def export():
    conn = sqlite3.connect(DB)
    df = pd.read_sql_query("SELECT * FROM fonctionnaires", conn)
    conn.close()
    df.to_excel("Etat_Salaires.xlsx", index=False)
    return send_file("Etat_Salaires.xlsx", as_attachment=True)

if __name__ == '__main__':
    init_db()
    app.run(debug=True)

**Comment by foudafoudaongolo-creator** [comment]
Cmd
